src/arrays.js

```diff
@@ -15,7 +21,19 @@ function sameFirst(a, b) {
  * @return {Boolean}
  */
 function deepEquals(a, b) {
+  for (var i = 0; i < a.length; i++) {
+
+    if (a[i] != b[i]) {
+      return false;
+    }
 
+    if (a.length != b.length) {
```

Comment:
This if statement can be done before the `for` loop so that you don't have to loop because if 2 arrays are different lengths there's no reason to even check if the first items are the same.